### ISSUE TITLE
Camera still in use after unmounting of component

### ISSUE THREAD

**Issue by rishabhgupta1234** [aws-amplify/amplify-ui#6245]
### Before creating a new issue, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-ui/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-ui/discussions).
- [X] I have tried disabling all browser extensions or using a different browser
- [X] I have tried deleting the node_modules folder and reinstalling my dependencies
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md#bug-reports).

### On which framework/platform are you having an issue?

React

### Which UI component?

Liveness

### How is your app built?

Create React App

### What browsers are you seeing the problem on?

Chrome, Firefox, Microsoft Edge

### Please describe your bug.

I am encountering an issue with **FaceLivenessDetectorCore** while implementing face liveness detection for users. After the face liveness process is completed, the camera light remains on, indicating that the camera is not being released properly.

It seems that **FaceLivenessDetectorCore** continues to access the user's camera permissions and does not release the camera resource once the face liveness process is finished.

### What's the expected behaviour?

After the face liveness process is completed, the camera should be released, and the camera light should turn off.

### Help us reproduce the bug!

1) Run FaceLivenessDetectorCore independently, and it will request camera permissions.
2) Complete the face liveness check.
3) After the process is completed, the camera light remains on.

### Code Snippet

```typescript

import { FaceLivenessDetectorCore } from '@aws-amplify/ui-react-liveness'

<FaceLivenessDetectorCore
    sessionId={sessionId}
    region={AWS_AMPLIFY_REGION}
    onAnalysisComplete={handleAnalysisComplete}
    onError={handleLivenessComponentError}
    config={getLivenessConfig()}
/>

```

**Comment by cwomack** [comment]
Hello, @rishabhgupta1234 and thank you for opening this issue.  We'll investigate this further and reproduce on our side.  If there's further questions or context needed, we'll follow up.  Thanks!

**Comment by rishabhgupta1234** [comment]
Hi @cwomack thanks for replying

Please find the details below

1) Yes, about webcam light
2) @aws-amplify/ui-react-liveness version is 3.1.1
3) Ubuntu 24.04.1

**Comment by rishabhgupta1234** [comment]
@dindjarinjs I'm rendering the Faceliveness component based on a toggle condition. After this step is completed, the video is not rendered in the DOM, but the webcam light remains on.

**Comment by cwomack** [comment]
@rishabhgupta1234, we've been unable to reproduce on our side.  Do you have a sample repo you can share where this is easily reproducible? Or could you share more of the code related to the toggle condition for us to have a better idea of how to reproduce? Thanks.

**Comment by rishabhgupta1234** [comment]
@cwomack Below link contains the code of the toggle condition

https://smalldev.tools/share-bin/e7KeCH5o

**Comment by cwomack** [comment]
@rishabhgupta1234, we've tried to reproduce this on our side but have had no luck.  Do you have minimal sample repo that can reliably reproduce the issue that you can share?

**Comment by rishabhgupta1234** [comment]
@cwomack Can you please share a sample repository that I can refer to and tweak according to my requirements

**Comment by rishabhgupta1234** [comment]
@cwomack Please find the sample GitHub repo to replicate the issue

https://github.com/rishabhgupta1234/face-liveliness-repo.git

**Comment by cwomack** [comment]
@rishabhgupta1234, appreciate you providing that! Will look into it and follow up.

**Comment by surpol** [comment]
I am also having the same issues on the M3 MacBook Pro for both Chrome and Safari. @cwomack 

package.json:
`    "@aws-amplify/ui-react-liveness": "^3.3.2"`

**Comment by HuiSF** [comment]
Hi @surpol thanks for following up on this issue. Looking at your code, you are simply unmount the UI component from the DOM tree. Following this I can reproduce the issue, that even though the `FaceLivenessDetector` is unmounted, the camera is still in use. I believe some clean up process is missing here. I will keep investigating.

**Comment by surpol** [comment]
Yes, please follow up

**Comment by HuiSF** [comment]
With further testing and checking the code around component unmounting, I noticed this unexpected behavior happens only with the React strict mode enabled, once I disable the strict mode, when unmounting the Liveness component, the camera stopped being used.

Hi @rishabhgupta1234 and @surpol could you confirm whether this issue happened for you with the React strict mode, too?

**Comment by surpol** [comment]
@HuiSF Yes, with React Strict Mode

`
        <React.StrictMode>
            <FaceLivenessDetector 
                sessionId={sessionId} 
                region="us-east-1"
                onAnalysisComplete={(result) => handleAnalysisComplete(sessionId, root)}
                onUserCancel={() => onUserCancel(root)}
                disableStartScreen={true}
                onError={(error) => handleError(error, root)}
                style={{
                    margin: 0,
                    padding: 0,
                    display: 'block',
                    lineHeight: 0
                }}
            />
        </React.StrictMode>
`

**Comment by HuiSF** [comment]
Thanks @surpol we will looking into a solution to make this component fully working with the React `StrictMode`, in the meantime, you can use the component without the `StrictMode` to get around of this issue.
